### ISSUE TITLE
TFormColorPicker color component issue / bug.

### ISSUE THREAD

**Issue by tednilsen** [graphics32/graphics32#131]
### GR32_Dsgn_ColorPicker.pas
The TFormColorPicker has an annoying bug, when sliding the RGBA pickers all the other components resets to $00.
This solves the issue / bug - the alpha component is now also preserved.
https://github.com/graphics32/graphics32/blob/8e45dbc50751fd324d948668209dfb4085435fe0/Source/GR32_Dsgn_ColorPicker.pas#L134-L137
replace with:
```pascal
procedure TFormColorPicker.ColorPickerChanged(Sender: TObject);
begin
  if (Sender = ColorPickerGTK)
    then Color := ColorPickerGTK.SelectedColor or ColorPickerAlpha.SelectedColor
    else Color := ColorPickerRed.SelectedColor or ColorPickerGreen.SelectedColor or
                  ColorPickerBlue.SelectedColor or ColorPickerAlpha.SelectedColor;
end;
```

https://github.com/graphics32/graphics32/blob/8e45dbc50751fd324d948668209dfb4085435fe0/Source/GR32_Dsgn_ColorPicker.pas#L184-L217
replace with:
```pascal
begin
  // disable OnChange handler
  EditColor.OnChange := nil;
  SpinEditRed.OnChange := nil;
  SpinEditGreen.OnChange := nil;
  SpinEditBlue.OnChange := nil;
  SpinEditAlpha.OnChange := nil;
  ColorPickerGTK.OnChanged := nil;

  // update spin edits
  Color32ToRGBA(FColor, R, G, B, A);
  SpinEditRed.Value := R;
  SpinEditGreen.Value := G;
  SpinEditBlue.Value := B;
  SpinEditAlpha.Value := A;

  // update color edit
  SelStart := EditColor.SelStart;
  EditColor.Text := '#' + IntToHex(A, 2) + IntToHex(R, 2) + IntToHex(G, 2) + IntToHex(B, 2);
  EditColor.SelStart := SelStart;

  ColorPickerRed.SelectedColor := FColor and $00FF0000;
  ColorPickerGreen.SelectedColor := FColor and $0000FF00;
  ColorPickerBlue.SelectedColor := FColor and $000000FF;
  ColorPickerAlpha.SelectedColor := FColor and $FF000000;
  ColorSwatch.Color := FColor;
  ColorPickerGTK.SelectedColor := Color32(R, G, B, 0);

  // re-enable OnChange handler
  ColorPickerGTK.OnChanged := ColorPickerChanged;
  SpinEditRed.OnChange := SpinEditColorChange;
  SpinEditGreen.OnChange := SpinEditColorChange;
  SpinEditBlue.OnChange := SpinEditColorChange;
  SpinEditAlpha.OnChange := SpinEditColorChange;
  EditColor.OnChange := EditColorChange;
end;
```

**Comment by micha137** [comment]
@tednilsen Thank you, but could you file a pull request?  Viewing a diff makes it easier for us to review and integrate your changes.

**Comment by tednilsen** [comment]
I think I did it correctly and issued / pushed a patched, you can delete this "fix" as the other one is complete.

**Comment by andersmelander** [comment]
You should have closed this one instead and left the pull request open. We can't merge your fix from the above commit as it doesn't belong to any existing branch.

**Comment by andersmelander** [comment]
FYI, I've pushed a minor improvement to this fix: [0c39bd7](https://github.com/graphics32/graphics32/commit/0c39bd7c937399cb1fdb2f54e06bad15c0013413)
